src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.c

```diff
@@ -174,7 +174,7 @@ static void on_hostbyname_done_cb(void *arg, int status, int timeouts,
           grpc_lb_addresses_set_address(
               *lb_addresses, i, &addr, addr_len,
               hr->is_balancer /* is_balancer */,
-              hr->is_balancer ? strdup(hr->host) : NULL /* balancer_name */,
+              hr->is_balancer ? hr->host : NULL /* balancer_name */,
```

Comment:
If this PR is not going to be merged fairly soon, perhaps we should split the changes from this file off into a separate PR, just to get this fix in?

Reply:
yeah, will open separate pr for that shortly